var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -481,6 +488,11 @@ def determine_b2_options(self, spec, options):
             raise RuntimeError("At least one of {singlethreaded, " +
                                "multithreaded} must be enabled")
 
+        # If we are building context using a non-fcontext backend, we need to tell b2
+        context_impl = spec.variants['context-impl'].value
+        if '+context' in spec and context_impl in {'ucontext', 'winfib'}:
+            options.extend(['context-impl=%s' % context_impl])
+
```

Comment:
Assuming b2 won't raise any fuss if we pass `context-impl=fcontext`, there's no need to check for it here. That will remove this maintenance hazard in keeping the check here and the values of the variant in synch.

```suggestion
if '+context' in spec:
  context_impl = spec.variants['context-impl'].value
  options.append('context-impl=%s' % context_impl)
```

Reply:
The documentation page I linked above only discusses the possibility of `ucontext` or `winfib` as arguments.

Reply:
In 1.79.0,

``` console
$ ./bootstrap.sh --with-libraries=context

$ ./b2 context-impl=moo
/boost_1_79_0/tools/build/src/build/feature.jam:491: in validate-value-string from module feature
error: "moo" is not a known value of feature <context-impl>
error: legal values: "fcontext" "ucontext" "winfib"

$ ./b2 context-impl=fcontext
  ...
    - context                  : building
  ...
The Boost C++ Libraries were successfully built!
```

Let's assume for now that always works. I'll add it to my TODO list to double-check.

Reply:
Cool - i just verified `fcontext` is present as an option at least as far back as 1.65:
https://github.com/boostorg/context/blob/boost-1.65.0/build/Jamfile.v2

I'll go ahead and remove that extra check.

I will note that the special-casing for the defines introduced in [65d7b34](https://github.com/spack/spack/pull/30496/commits/65d7b34ce1839e576f377bba64a5eb9c256a2a58) should be kept as `BOOST_USE_FCONTEXT` is not used anywhere.

Reply:
Oh, that's... neat. Particularly since ucontext was [removed](https://www.boost.org/doc/libs/1_79_0/libs/context/doc/html/context/rationale/other_apis_.html#ucontext) in POSIX.1-2008. Boost, why did you do that?!

Out of curiosity, which backend are you using?

Reply:
fcontext is definitely the sensible default for most use-cases, but ucontext is the only way to cooperate with ASAN if you're doing memory debugging (or was - it seems to have broken at some point recently since ASAN promptly reported a bug in boost itself when I switched)

Reply:
Oh sorry, yes, you mentioned that in your commit. It's interesting ASAN is relying on the deprecated/removed interface. I would assume it's really difficult to instrument the fcontext usage.